message-index/messages/GHC-64725/display/index.md

```diff
@@ -0,0 +1,37 @@
+---
+title: Cannot Display functions
+---
+
+## Error Message
+
+```
+Main2.hs:12:23: error: [GHC-64725]
+    • Cannot Display functions.
+      Perhaps there is a missing arguments?
+    • In the expression: display myFunction
+      In an equation for ‘myDisplayedFunction’:
+          myDisplayedFunction = display myFunction
+   |
+12 | myDisplayedFunction = display myFunction
+   |                       ^^^^^^^
+```
+
+## Explanation
+
+Here, the module `SomeDisplayLibrary` stands in for a pretty-printing library,
+used to produce human readable text from Haskell values. The library provides a
+type class `Display` with instances for different types, and the function
+`display :: Display a => a -> String` to perform the conversion. In this case
+the result is of type `String` but some libraries use their own data type to
+structure the output.
+
+A problem common to these libraries is that of function typed values, since
+there's no general way to represent them as strings. Perhaps for _some_
+functions one could simply write the mapping from inputs to outputs, if the input space is enumerable and small enough, but it becomes impractical for larger types, and unfeasible for non-enumerable types, so usually these libraries don't provide an instance for functions. In this case, the library
```

Comment:
```suggestion
there's no general way to represent them as strings. Perhaps for _some_
functions one could simply write the mapping from inputs to outputs - if the input space is enumerable and small enough - but it becomes impractical for larger types, and unfeasible for non-enumerable types. Usually these libraries don't provide an instance for functions. In this case, the library
```

Less winding sentence

Reply:
heh yeah I've been known to write those. Thanks!